app/src/main/java/com/dankideacentral/dic/authentication/OAuthWebViewFragment.java

```diff
@@ -26,28 +29,31 @@
     public void onViewCreated(View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         String auth = getArguments().getString(getString(R.string.string_extra_authentication_url));
+        final CookieManager cookieManager = CookieManager.getInstance();
 
         webView.loadUrl(auth);
         webView.setWebViewClient(new WebViewClient()
         {
             @Override
             public boolean shouldOverrideUrlLoading(WebView view, String url) {
                 if (url.contains("oauth_verifier=")) {
-                    Intent intent;
 
                     // Create user session
                     TwitterSession.getInstance().createSession(Uri.parse(url),
-                            getActivity().getApplicationContext());
-
-                    if (isLocationEnabled() == PackageManager.PERMISSION_GRANTED) {
-                        intent = new Intent(getActivity().getApplicationContext(),
-                                TweetFeedActivity.class);
-                    } else {
-                        intent = new Intent(getActivity().getApplicationContext(),
-                                SearchActivity.class);
-                    }
-
-                    startActivity(intent);
+                            getActivity().getApplicationContext(),
+                            new Handler.Callback() {
```

Comment:
Thanks for telling me about the Callback, learned something new ahyuehehehehehehe